### ISSUE TITLE
Mermaid diagrams not rendering when page is titled Mermaid

### ISSUE THREAD

**Issue by FISHMANPET** [squidfunk/mkdocs-material#3652]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

Pages with the title of `Mermaid` or `mermaid` (as defined via the top level header in the md file) do not render mermaid diagrams. If the filename is `mermaid.md` and no header is defined in the file, the mermaid diagram will render.

### Expected behaviour

Mermaid diagrams render on pages titled "Mermaid"

### Actual behaviour

Mermaid diagrams do not render on pages titled "Mermaid"

### Steps to reproduce

I've created a minimal site that demonstrates the behavior: https://github.com/FISHMANPET/mkdocs-mermaid
I've also hosted it on a webserver, because when you view it that way (rather than `mkdocs serve`) you will see errors in the browser developer console as well: https://tst.docs.oit.umn.edu/mermaid/

### Package versions

- Python: `3.9.5`
- MkDocs: `1.2.3`
- Material: `8.2.3`

### Configuration

```yaml
# https://github.com/FISHMANPET/mkdocs-mermaid/blob/main/mkdocs.yml
site_name: My Docs
theme:
  name: material
markdown_extensions:
  - pymdownx.superfences:
      custom_fences:
        - name: mermaid
          class: mermaid
          format: !!python/name:pymdownx.superfences.fence_code_format
```

### System information

- Operating system: Windows 10
- Browser: Chrome 98

**Comment by squidfunk** [comment]
Thanks for reportinhg. Exact duplicate of #3596. Explanation and mitigation in https://github.com/squidfunk/mkdocs-material/issues/3596#issuecomment-1047160173

**Comment by squidfunk** [comment]
@FISHMANPET Funny. After I've [read your tweet](https://twitter.com/fishmanpet/status/1499028570626273281), I actually came up with a way to fix it, and it's quite simple. The reason is that while reading about this behavior, I realized that we could just add a simple check to mitigate the problem at hand, so it's fixable after all. Sometimes a little time to think about stuff or learn works wonders. Fixed in cfa64745a.
